websites/D/Dek-d/presence.ts

```diff
@@ -0,0 +1,307 @@
+const presence = new Presence({
+		clientId: "1314484095614451822",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/EUDvEz9.png",
+}
+
+presence.on("UpdateData", async () => {
+	const sections = {
+			quiz: [
+				"horoscope",
+				"habit",
+				"character",
+				"love",
+				"entertainment",
+				"joke",
+				"education",
+				"knowledge",
+				"adventure",
+				"music",
+				"movie",
+				"cartoon",
+				"game",
+				"novel",
+				"sport",
+				"pet",
+				"other",
+			],
+			tcas: [
+				"dreamcareer",
+				"reality",
+				"onstage",
+				"dreamcampus",
+				"ontour",
+				"review",
+			],
+			activity: "",
+			studyabroad: [
+				"exchange",
+				"high_school",
+				"bachelor",
+				"master",
+				"course",
+				"work_travel",
+				"foreign_culture",
+				"global_review",
+				"aec",
+				"english_issue",
+			],
+			nugirl: "",
+			teentrends: "",
+			"pre-admission": "",
+			studyabroadfair: "",
+			wallet: "",
+			board: [
+				"tcas",
+				"entertainment",
+				"knowledge",
+				"teen",
+				"nugirl",
+				"education",
+				"studyabroad",
+				"writer",
+				"problem",
+			],
+			tag: "",
+			visualnovel: "",
+			loveroom: "",
+			featured: "",
+			store: "",
+			review: "",
+			collection: "",
+		},
+		titleText = document.querySelector("h1.title")?.textContent ?? "",
+		pathSegments = location.pathname.split("/"),
+		presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+			type: ActivityType.Watching,
+			// initial details in case of no match displaying the 1st path segment
+			details: `Browsing ${pathSegments[1]}`,
+			buttons: [{ label: "Visit", url: document.URL }],
+		},
+		// board thread play function
+		displaySec = (categorys = " ") => {
+			presenceData.details = `Viewing${categorys}discussion thread`;
+		};
+
+	// main
+	switch (location.hostname) {
+		// handle subdomain ex novels.dek-d.com
+		case "www.dek-d.com": {
+			if (location.pathname === "/") presenceData.details = "Viewing home page";
+			else if (Object.keys(sections).includes(pathSegments[1])) {
+				switch (pathSegments[1]) {
+					case "board": {
+						displaySec(" ");
+						if (sections.board.includes(pathSegments[2]))
+							displaySec(` ${pathSegments[2]} `);
+
+						if (pathSegments[3] !== "") presenceData.state = `${titleText}`;
+
+						break;
+					}
+					case "tcas": {
+						presenceData.details = `Viewing ${pathSegments[1]} thread`;
+						if (!isNaN(Number(pathSegments[2])))
+							presenceData.state = `${titleText}`;
+						else if (sections.tcas.includes(pathSegments[2]))
+							presenceData.details = `Browsing ${pathSegments[2]} on ${pathSegments[1]} thread`;
+						presenceData.buttons = [{ label: "View", url: document.URL }];
```

Comment:
Use `document.location.href` instead of` document.URL`. Also consider using a more specific button label.